CMakeLists.txt

```diff
@@ -17,6 +17,9 @@ option(ADDRESS_SANITIZER "Build with address sanitizer" OFF)
 # vcpkg integration
 include(AzureVcpkg)
 az_vcpkg_integrate()
+set(AZ_PLATFORM_IMPL "POSIX" CACHE STRING "Platform")
```

Comment:
If we make AZ_PLATFORM_IMPL into `option` then we won't be able to set it to `"POSIX"`, `"WIN32"`, etc. since option will make it into a bool.

https://cmake.org/cmake/help/latest/command/option.html